thinc/schedules.py

```diff
@@ -1,32 +1,52 @@
 """Generators that provide different rates, schedules, decays or series."""
-from typing import Iterable
+from typing import Dict, Iterable, Optional, Tuple
+from typing_extensions import Protocol, runtime_checkable
 import numpy
 
 from .config import registry
 
 
+@runtime_checkable
+class ScheduleCallable(Protocol):
```

Comment:
Alternatively, this definition could go into `thinc.types`

Reply:
That's a better place, moved.